File: practise-1/bank_account.py


class BankAccount:
    def __init__(self, account_number, account_name, balance):
        self.account_number = account_number
        self.account_name = account_name
        self.balance = balance

    def withdraw(self, amount):
        if amount <= 0:
            raise ValueError("Withdraw amount must be positive.")
        if amount > self.balance:
            raise ValueError("Insufficient funds.")
        self.balance -= amount
        return self.balance

    def deposit(self, amount):
        if amount <= 0:
            raise ValueError("Deposit amount must be positive.")
        self.balance += amount
        return self.balance

    def transfer(self, recipient_account, amount):
        if amount <= 0:
            raise ValueError("Transfer amount must be positive.")
        if amount > self.balance:
            raise ValueError("Insufficient funds for transfer.")
        self.withdraw(amount)
        recipient_account.deposit(amount)
        return self.balance

File: practise-6/BankAccountApp/BankAccount.cs


class BankAccount {
    public string Name {set; get;}
    public string AccountNo {set; get;}
    public double Balance {private set; get;}

    public void Deposit(double amount){
        Balance = Balance + amount;
    }

    public void Withdraw(double amount){
        if (Balance - amount > 0){
            Balance = Balance - amount;
        }
        else {
            throw new ApplicationException("Insufficient balance!");
        }
    }
}